File: 프로그래머스/LV2/월간 코드 챌린지 시즌3/n^2 배열 자르기/n^2 배열 자르기.py
def solution(n, left, right):
    return [i//n + 1 if i // n > i % n else i % n + 1 for i in range(left, right + 1)]

# import numpy as np

# def solution(n, left, right):
#     answer = np.array([])
#     sr = left // n
#     row = np.array(range(1,n+1))
#     row[:sr] = row[sr]
#     answer = np.concatenate([answer, row])

#     for i in range(sr+1, sr + (right - left) // n + 2):
#         row[:i] += 1
#         answer = np.concatenate([answer, row])
#     return list(map(int, answer[left-sr*n:right-sr*n+1]))

File: 프로그래머스/LV2/코딩테스트 연습/월간 코드 챌린지 시즌3/n^2 배열 자르기/n^2 배열 자르기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n, long long left, long long right) 
{
    int* answer = (int*)malloc(sizeof(long long) * (right - left + 1));
    for(long long i = left; i <= right; i++)answer[i-left] = i / n > i % n ? i / n + 1 : i % n + 1;
    return answer;
}